fpga_interchange/arch.h

```diff
@@ -825,10 +823,9 @@ struct Arch : ArchAPI<ArchRanges>
             if (!is_cell_valid_constraints(cell, tile_status, explain_constraints)) {
                 return false;
             }
-
-            auto &bel_data = bel_info(chip_info, bel);
-            return get_site_status(tile_status, bel_data).checkSiteRouting(getCtx(), tile_status);
         }
+        auto &bel_data = bel_info(chip_info, bel);
+        return get_site_status(tile_status, bel_data).checkSiteRouting(getCtx(), tile_status);
```

Comment:
Need a comment here about why we are checking site status for an unbound BEL.  I need to check that the SiteRouter does the right thing on an empty site.